### PR TITLE
MM-58793 Apply EnableUserCreation to signup.tsx

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Clarifies the `EnableUserCreation` and `EnableOpenServer` settings and when they are applied. Also, applies the `EnableUserCreation` setting to the `signup.tsx` page. Previously, the user would signup (Email) or login(LDAP) before the server would fail to create a user.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-58793

#### Screenshots
Before - Email and Google(LDAP) signups would fail at server.

Screenshot 2024-07-17 at 3 50 49 PM

After - Email/LDAP signups no longer available
Screenshot 2024-07-17 at 3 49 33 PM

#### Release Note

```release-note
Applied `EnableUserCreation` to the UI Signup page, to prevent appearance of ability in UI. 
```

### CODE REVIEW

**Comment by sbishel** [comment]
@cwarnermm - I put you as the first reviewer. Can you help me with the help text? I'm sure you will have something better than mine. Also once we determine the proper text, it should be updated here as well.
https://docs.mattermost.com/configure/authentication-configuration-settings.html#signup

File: webapp/channels/src/components/admin_console/admin_definition.tsx [hunk 1]
```diff
@@ -3046,7 +3046,7 @@ const AdminDefinition: AdminDefinitionType = {
                             type: 'bool',
                             key: 'TeamSettings.EnableUserCreation',
                             label: defineMessage({id: 'admin.team.userCreationTitle', defaultMessage: 'Enable Account Creation: '}),
-                            help_text: defineMessage({id: 'admin.team.userCreationDescription', defaultMessage: 'When false, the ability to create accounts is disabled. The create account button displays error when pressed.'}),
+                            help_text: defineMessage({id: 'admin.team.userCreationDescription', defaultMessage: 'When false, the ability to create accounts is disabled. The create account button displays error when pressed. Applies to email/OpenID/OAuth accounts.'}),
```
Comment (cwarnermm) [inline webapp/channels/src/components/admin_console/admin_definition.tsx:3049]:
```suggestion
                            help_text: defineMessage({id: 'admin.team.userCreationDescription', defaultMessage: 'When false, the ability to create accounts is disabled, and selecting Create Account displays an error. Applies to Email, OpenID Connect, and OAuth 2.0 user account authentication.'}),
```

File: webapp/channels/src/components/admin_console/admin_definition.tsx [hunk 2]
```diff
@@ -3077,7 +3077,7 @@ const AdminDefinition: AdminDefinitionType = {
                             type: 'bool',
                             key: 'TeamSettings.EnableOpenServer',
                             label: defineMessage({id: 'admin.team.openServerTitle', defaultMessage: 'Enable Open Server: '}),
-                            help_text: defineMessage({id: 'admin.team.openServerDescription', defaultMessage: 'When true, anyone can signup for a user account on this server without the need to be invited.'}),
+                            help_text: defineMessage({id: 'admin.team.openServerDescription', defaultMessage: 'When true, anyone can signup for a user account on this server without the need to be invited. Applies to Email signups only.'}),
```
Comment (cwarnermm) [inline webapp/channels/src/components/admin_console/admin_definition.tsx:3080]:
```suggestion
                            help_text: defineMessage({id: 'admin.team.openServerDescription', defaultMessage: 'When true, anyone can sign up for a user account on this server without the need to be invited. Applies to Email-based signups only.'}),
```

File: webapp/channels/src/i18n/en.json [hunk 3]
```diff
@@ -2660,7 +2660,7 @@
   "admin.team.maxUsersExample": "E.g.: \"25\"",
   "admin.team.maxUsersTitle": "Max Users Per Team:",
   "admin.team.noBrandImage": "No brand image uploaded",
-  "admin.team.openServerDescription": "When true, anyone can signup for a user account on this server without the need to be invited.",
+  "admin.team.openServerDescription": "When true, anyone can signup for a user account on this server without the need to be invited. Applies to email signups only.",
```
Comment (cwarnermm) [inline webapp/channels/src/i18n/en.json:2663]:
```suggestion
  "admin.team.openServerDescription": "When true, anyone can sign up for a user account on this server without the need to be invited. Applies to Email-based signups only.",
```

File: webapp/channels/src/i18n/en.json [hunk 4]
```diff
@@ -2683,7 +2683,7 @@
   "admin.team.teammateNameDisplay": "Teammate Name Display:",
   "admin.team.teammateNameDisplayDesc": "Set how to display users' names in posts and the Direct Messages list.",
   "admin.team.uploadDesc": "Customize your user experience by adding a custom image to your login screen. Recommended maximum image size is less than 2 MB.",
-  "admin.team.userCreationDescription": "When false, the ability to create accounts is disabled. The create account button displays error when pressed.",
+  "admin.team.userCreationDescription": "When false, the ability to create accounts is disabled. The create account button displays error when pressed. Applies to Email/OpenID/OAuth accounts.",
```
Comment (cwarnermm) [inline webapp/channels/src/i18n/en.json:2686]:
```suggestion
  "admin.team.userCreationDescription": "When false, the ability to create accounts is disabled, and selecting Create Account displays an error. Applies to Email, OpenID Connect, and OAuth 2.0 user account authentication.",
```

**Review by cwarnermm — COMMENTED** [review]
Provided updated in-product help text inline CC @sbishel

**Review by calebroseland — CHANGES_REQUESTED** [review]
Left a comment on the ticket: https://mattermost.atlassian.net/browse/MM-58793?focusedCommentId=168017

**Review by JulienTant — CHANGES_REQUESTED** [review]
Requesting change for the same reason @calebroseland highlighted; but the code changed here LGTM.

**Review by JulienTant — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]

File: webapp/channels/src/components/admin_console/admin_definition.tsx [hunk 5]
```diff
@@ -3046,7 +3046,7 @@ const AdminDefinition: AdminDefinitionType = {
                             type: 'bool',
                             key: 'TeamSettings.EnableUserCreation',
                             label: defineMessage({id: 'admin.team.userCreationTitle', defaultMessage: 'Enable Account Creation: '}),
-                            help_text: defineMessage({id: 'admin.team.userCreationDescription', defaultMessage: 'When false, the ability to create accounts is disabled. The create account button displays error when pressed.'}),
+                            help_text: defineMessage({id: 'admin.team.userCreationDescription', defaultMessage: 'When false, the ability to create accounts is disabled, and selecting Create Account displays an error. Applies to Email, OpenID Connect, and OAuth 2.0 user account authentication.'}),
```
Comment (esarafianou) [inline webapp/channels/src/components/admin_console/admin_definition.tsx:3049]: So this setting doesn't apply to LDAP and SAML, right? It's clear on the text, just verifying that this is the case in the code.

**Review by esarafianou — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - I will have to test this post merge as, on the spinwicks, "Enable Open Server" setting is set to "true" through an environment variable and I'm not able to change it.

File: webapp/channels/src/components/admin_console/admin_definition.tsx (hunk 5 above)
Comment (sbishel) [reply webapp/channels/src/components/admin_console/admin_definition.tsx:3049]: Correct, LDAP and SAML users can always create an account by logging in via LDAP, SAML.

**Pull request merged**